tests/lib/cli-engine/cli-engine.js

```diff
@@ -3709,46 +3896,66 @@ describe("CLIEngine", () => {
         describe("don't ignore the entry directory.", () => {
             const root = getFixturePath("cli-engine/dont-ignore-entry-dir");
 
-            it("'executeOnFiles(\".\")' should not load config files from outside of \".\".", () => {
-                CLIEngine = defineCLIEngineWithInMemoryFileSystem({
-                    cwd: () => root,
+            /* eslint-disable-next-line func-style */
+            let cleanup = () => { };
+
+            afterEach(async () => {
+                await cleanup();
+                cleanup = () => { };
+            });
+
+            it("'executeOnFiles(\".\")' should not load config files from outside of \".\".", async () => {
+                const teardown = createCustomTeardown({
+                    cwd: root,
                     files: {
                         "../.eslintrc.json": "BROKEN FILE",
                         ".eslintrc.json": JSON.stringify({ root: true }),
                         "index.js": "console.log(\"hello\")"
                     }
-                }).CLIEngine;
-                engine = new CLIEngine();
+                });
+
+                engine = new CLIEngine({ cwd: teardown.getPath() });
+
+                await teardown.prepare();
+                cleanup = teardown.cleanup;
 
                 // Don't throw "failed to load config file" error.
                 engine.executeOnFiles(".");
             });
 
-            it("'executeOnFiles(\".\")' should not ignore '.' even if 'ignorePatterns' contains it.", () => {
-                CLIEngine = defineCLIEngineWithInMemoryFileSystem({
-                    cwd: () => root,
+            it("'executeOnFiles(\".\")' should not ignore '.' even if 'ignorePatterns' contains it.", async () => {
+                const teardown = createCustomTeardown({
+                    cwd: root,
                     files: {
-                        "../.eslintrc.json": JSON.stringify({ ignorePatterns: ["/dont-ignore-entry-dir"] }),
-                        ".eslintrc.json": JSON.stringify({ root: true }),
+                        "../.eslintrc.json": { ignorePatterns: ["/dont-ignore-entry-dir"] },
```

Comment:
Same as above, this file will probably remain after the test.

Reply:
Same, this file is safely removed.

Reply:
Hm, I tried with commenting out this: 

https://github.com/eslint/eslint/blob/1df7fa69967cc514a541c363eb52d845e1b1d8f0/tests/lib/cli-engine/cli-engine.js#L92-L94

and the file was still there, so it looks like it remains undeleted between tests.

I think the lib removes only the innermost directory that was initally passed as `baseDir`:

https://github.com/kettanaito/fs-teardown/blob/b801c07f764563cd413685c523dd9319247c720e/src/createTeardown.ts#L80-L82

Reply:
Another way to check, try to completely remove this test and then run all tests:

https://github.com/eslint/eslint/blob/db6787f7478772d1c26d93fbba041dcde4a23585/tests/lib/cli-engine/cli-engine.js#L3928-L3945

If we just remove this test (or change the order of tests), the invalid config file `"../.eslintrc.json": "BROKEN FILE"` from the previous test will not be overwritten with a valid one from this test, so all of the remaining tests will fail because they'll have a broken config file in an ancestor directory which wasn't cleaned up.